### PR TITLE
v4.2.2. Do not do scribunto user read checks when in CLI mode. Mainte…

### PR DESCRIPTION
**Pull request by Designburo**
…nance scripts will never be able to parse a page as the user never has rights.

### CODE REVIEW

File: src/Scribunto/ScribuntoLuaLibrary.php [hunk 1]
```diff
@@ -33,6 +33,28 @@ public function register(): void {
 		$this->getEngine()->registerInterface( __DIR__ . '/' . 'mw.wsslots.lua', $interfaceFuncs, [] );
 	}
 
+	/**
+	 * @param \WikiPage $wikiPage
+	 *
+	 * @return bool
+	 */
+	private function userCan( \WikiPage $wikiPage ): bool {
```
Comment (marijnvanwezel) [inline src/Scribunto/ScribuntoLuaLibrary.php:41]: Use a `use` statement to import WikiPage and omit the global namespace prefix (`\`).

File: src/Scribunto/ScribuntoLuaLibrary.php [hunk 2]
```diff
@@ -33,6 +33,28 @@ public function register(): void {
...
+		// Only do a check for user rights when not in cli mode
+		if ( PHP_SAPI != 'cli' ) {
```
Comment (marijnvanwezel) [inline src/Scribunto/ScribuntoLuaLibrary.php:43]: Use strict equality.

Comment (marijnvanwezel) [inline src/Scribunto/ScribuntoLuaLibrary.php:43]: Reverse the `if` statement expression and use a guard-clause instead. That is, have the entire `if` block be:

```php
if ( PHP_SAPI === 'cli' ) {
    return true;
}
```

File: src/Scribunto/ScribuntoLuaLibrary.php (hunk 1 above)
Comment (marijnvanwezel) [inline src/Scribunto/ScribuntoLuaLibrary.php:41]: Please place all `private` methods below `public` methods.

File: src/Scribunto/ScribuntoLuaLibrary.php [hunk 3]
```diff
@@ -33,6 +33,28 @@ public function register(): void {
...
+			$userCan = MediaWikiServices::getInstance()->getPermissionManager()->userCan(
+				'read',
+				RequestContext::getMain()->getUser(),
+				$wikiPage->getTitle()
+			);
+
+			if ( !$userCan ) {
```
Comment (marijnvanwezel) [inline src/Scribunto/ScribuntoLuaLibrary.php:50]: Can be simplified to `return $userCan;` (when taking into account other comments).

**Review by marijnvanwezel — CHANGES_REQUESTED** [review]

**Comment by marijnvanwezel** [comment]
I am also wondering whether this might be an issue for other `read`-checks that WSSlots does, such as in parser functions.

**Comment by Designburo** [comment]
Those are mw defined code changes?

**Comment by marijnvanwezel** [comment]
I am fine with most of the changes, but changes like putting the arguments of `array_unshift` on multiple lines is just ugly:

```php
array_unshift(
    $array,
    ''
);
```

`array_unshift( $array, '' );` is much more readable. It seems that MediaWiki CodeSniffer configuration doesn't care about these kind of style changes, as running `composer fix` on either version returns "No fixable errors".

That being said, I like to keep pull requests as small as possible. I will do some git magic to undo the styling changes outside of the function.

**Pull request merged**